ipset.go

```diff
@@ -92,34 +105,49 @@ func (s *IPSet) RemovePrefix(p IPPrefix) { s.RemoveRange(p.Range()) }
 
 // RemoveRange removes r from s.
 func (s *IPSet) RemoveRange(r IPRange) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.removeRangeLocked(r)
+}
+
+func (s *IPSet) removeRangeLocked(r IPRange) {
 	if r.Valid() {
 		s.out = append(s.out, r)
+		s.normalized = false
 	}
 }
 
 // AddSet adds all ranges in b to s.
 func (s *IPSet) AddSet(b *IPSet) {
-	for _, r := range b.Ranges() {
-		s.AddRange(r)
+	rr := b.Ranges()
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	for _, r := range rr {
+		s.addRangeLocked(r)
 	}
 }
 
 // RemoveSet removes all ranges in b from s.
 func (s *IPSet) RemoveSet(b *IPSet) {
-	for _, r := range b.Ranges() {
-		s.RemoveRange(r)
+	rr := b.Ranges()
```

Comment:
Same—rangesLocked?

Reply:
Same as above, lock ordering makes this tricky.